cmd/az-analyze/cmd/common.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"path/filepath"
+	azDiskClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+)
+
+var driverNamespace string = viper.GetString("driverNamespace")
+
+// access to config
+func getConfig() *rest.Config {
+	kubeconfig := viper.GetString("kubeconfig")
+
+	if home, _ := homedir.Dir(); home != "" {
```

Comment:
Since we should give precedence to the params present in the config file, we should do this part only if kubeConfig=="". Otheriwse we'll end up overwriting the kubeconfig setup by the user in the config file.